inlong-manager/manager-dao/src/main/resources/mappers/DataNodeEntityMapper.xml

```diff
@@ -212,5 +212,55 @@
             </if>
         </where>
     </delete>
-
+    <insert id="copy">
+        insert into data_node (name, display_name, type, url, username,
+                               token, ext_params, description, tenant,
+                               in_charges, status, creator, modifier)
+        select #{newName, jdbcType=VARCHAR},
+               display_name,
+               type,
+               url,
+               username,
+               token,
+               ext_params,
+               description,
+               #{to, jdbcType=VARCHAR},
+               in_charges,
+               status,
+               creator,
+               modifier
+        from data_node
+        <where>
+            name = #{name, jdbcType=VARCHAR}
+            and type = #{type, jdbcType=VARCHAR}
+            and is_deleted = 0
+            and tenant = #{from, jdbcType=VARCHAR}
+        </where>
+    </insert>
+    <select id="selectByIdSelective"
+            parameterType="org.apache.inlong.manager.dao.entity.DataNodeEntity"
+            resultMap="BaseResultMap">
+        select
+        <include refid="Base_Column_List"/>
+        from data_node
+        <where>
+            <if test="type != null">
```

Comment:
`<if test="type != null and type != ''">`

Reply:
No need to do that,  it's insert, not update.